streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -519,6 +563,33 @@ private void handleTasksPendingInitialization() {
         }
     }
 
+    private void handleInitialTaskReuse(final Map<TaskId, Set<TopicPartition>> activeTasksToCreate,
+                                        final Map<TaskId, Set<TopicPartition>> standbyTasksToCreate,
+                                        final Map<TaskId, RuntimeException> failedTasks) {
+        final Map<Task, Set<TopicPartition>> initialStandbyTasksToRecycle = assignInitialTasks(activeTasksToCreate, logPrefix, topologyMetadata, changelogReader);
+        final Map<Task, Set<TopicPartition>> initialStandbyTasksToUse = assignInitialTasks(standbyTasksToCreate, logPrefix, topologyMetadata, changelogReader);
+
+        // if this was the last local thread to receive its assignment, close all the remaining Tasks, as they are not needed
+        stateDirectory.closeInitialTasksIfLastAssginedThread();
+
+        // recycle the initial standbys to active, and remove them from the set of actives that need to be created
+        if (!initialStandbyTasksToRecycle.isEmpty()) {
```

Comment:
Do we need this? The loop would not execute anyway if empty...

Reply:
True. I just included it to avoid allocating the `HashSet` on the next line if we're not going to use it anyway. Shouldn't be a big deal either way.

Reply:
Seems we optimize for different thing when writing code :) -- To me, it's mental burden to the reader so I try to avoid unnecessary optimization like this (as it won't make a any perf difference), in favor of fewer code to read and understand. It's another branch/condition I need to think about.

Leave it up to you to keep or remove.

Reply:
It looks like I actually had a bug here, which I believe I've fixed in https://github.com/apache/kafka/pull/16922/commits/d326ea8142629200ab15c7f38a45e75e95738edd

This makes the if-statement a little less pointless, but only marginally! I'm choosing to keep it, mainly because it helps to demarcate a "scope" for a specific piece of behaviour (i.e. what to do when we have some standbys to recycle).

Reply:
As you said -- matter of test. Ok as-is.